cmd/kubelet/app/server.go

```diff
@@ -418,11 +425,10 @@ func InitializeTLS(s *options.KubeletServer) (*server.TLSOptions, error) {
 	if s.TLSCertFile == "" && s.TLSPrivateKeyFile == "" {
 		s.TLSCertFile = path.Join(s.CertDirectory, "kubelet.crt")
 		s.TLSPrivateKeyFile = path.Join(s.CertDirectory, "kubelet.key")
-		if crypto.ShouldGenSelfSignedCerts(s.TLSCertFile, s.TLSPrivateKeyFile) {
-			if err := crypto.GenerateSelfSignedCert(nodeutil.GetHostname(s.HostnameOverride), s.TLSCertFile, s.TLSPrivateKeyFile, nil, nil); err != nil {
-				return nil, fmt.Errorf("unable to generate self signed cert: %v", err)
+		if !crypto.FoundCertOrKey(s.TLSCertFile, s.TLSPrivateKeyFile) {
```

Comment:
if we're in bootstrap mode, we probably also want to make sure the cert contains accurate SANs for the current hostname and IP... if I restart a node and it's IP has changed, or I change the hostname override, I would expect it to go get a new serving cert if it is in bootstrap mode